### PR TITLE
KUBOS-187 Adding Debug Configuration for the stm32f407 discovery target

### PR DESCRIPTION
**Pull request by kyleparrott**
Very basic debugging implementation for the stm32f4 discovery board.

I'm not crazy about the debugging implementation, specifically how openocd is started and not managed very well, or really at all by this implementation. It does work but should definitely be improved soon.

### CODE REVIEW

**Comment by kyleparrott** [comment]
In the future I want to transition this to some sort of python script that uses [pyOCD](https://pypi.python.org/pypi/pyOCD/0.5.1) or something similar to manage the openOCD gdb server.

File: targets/target-stm32f407-disco-gcc/openocd/flash.sh [hunk 1]
```diff
@@ -1,27 +1,40 @@
 #!/bin/bash
 
 this_dir=$(cd "`dirname "$0"`"; pwd)
-cmd=$1
+program=$1 # program is fist becasue for debugging there is not a command provided
```
Comment (davesims) [inline targets/target-stm32f407-disco-gcc/openocd/flash.sh]: "first because" :bowtie:

**Comment by kyleparrott** [comment]
Clearly I need to follow: https://www.linux.com/learn/using-spell-checking-vim 😄

File: targets/target-stm32f407-disco-gcc/openocd/flash.sh [hunk 2]
```diff
@@ -1,27 +1,41 @@
...
+cmd=$2
+if [[ ! -z $cmd ]]; then
+    openocd_arg="$cmd $program"
+fi
 unamestr=`uname`
 
 if [[ "$unamestr" =~ "Linux" ]]; then
-	device=`lsusb -d '0483:'`
+    device=`lsusb -d '0483:'`
 elif [[ "$unamestr" =~ "Darwin" ]]; then
```
Comment (plauche) [inline targets/target-stm32f407-disco-gcc/openocd/flash.sh]: nit: I think we can safely remove this OSX section, as this script is now targeted at running specifically in our vm.

Comment (kyleparrott) [inline targets/target-stm32f407-disco-gcc/openocd/flash.sh]: I still use this natively on my mac, even if just for testing I would like to request that we leave it in there.

**Comment by kyleparrott** [comment]
@davesims: friendly reminder to take another look at this PR.

**Comment by kyleparrott** [comment]
@davesims: Another Friendly bump to take a look at this.

**Comment by davesims** [comment]
whoops, sorry for the delay on this @kyleparrott.

**Pull request merged**